rct229/rules/section5/section5rule28.py

```diff
@@ -0,0 +1,139 @@
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                each_rule=Section5Rule28.BuildingRule.UnregulatedSurfaceRule(),
+                index_rmr="baseline",
+            )
+
+        def create_data(self, context, data=None):
+            building = context.baseline
+            # Merge into the existing data dict
+            return {
+                **data,
+                "surface_conditioning_category_dict": get_surface_conditioning_category_dict(
+                    data["climate_zone"], building
+                ),
+            }
+
+        def create_context_list(self, context, data=None):
+            # List of all baseline unregulated surfaces to become the context for Unregulated Surfaces
+            scc = data["surface_conditioning_category_dict"]
+
+            baseline_surfaces = find_all("$..surfaces[*]", context.baseline)
+            proposed_surfaces = find_all("$..surfaces[*]", context.proposed)
+
+            # This assumes that the surfaces matched by IDs between proposed and baseline
+            matched_proposed_surfaces = match_lists_by_id(
+                baseline_surfaces, proposed_surfaces
+            )
+
+            proposed_baseline_surface_pairs = zip(
+                baseline_surfaces, matched_proposed_surfaces
+            )
+
+            return [
+                UserBaselineProposedVals(None, surface_b, surface_p)
+                for surface_b, surface_p in proposed_baseline_surface_pairs
+                if scc[surface_b["id"]] == SCC.UNREGULATED
+            ]
+
+        class UnregulatedSurfaceRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(
+                    Section5Rule28.BuildingRule.UnregulatedSurfaceRule, self
+                ).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, True),
+                    list_path="subsurfaces[*]"
+                )
+
+            class UnregulatedSubsurfaceRule(RuleDefinitionBase):
+                def __init__(self):
+                    super(
+                        Section5Rule28.BuildingRule.UnregulatedSurfaceRule.UnregulatedSubsurfaceRule,
+                        self,
+                    ).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, True),
+                        required_fields={
+                            "$": [
+                                "u_factor",
+                                "solar_heat_gain_coefficient",
+                                "glazed_area",
+                                "opaque_area",
+                            ]
+                        },
+                    )
+
+                def get_calc_vals(self, context, data=None):
+                    subsurface_b = context.baseline
+                    subsurface_p = context.proposed
+
+                    return {
+                        "baseline_subsurface_u_factor": subsurface_b["u_factor"],
```

Comment:
I'm a little disoriented. Do all UNREGULATED subsurfaces have to have `u_factor`, `shgc`, `glazed_area`, and `opaque_area`? I can imagine that **if they have them**, they must be the same.  

Brainstorming:  If we use get() for all these, then we could eliminate these as required_fields, but `std_equal()` in the rule_check doesn't work with None arguments. Hmm.

Reply:
hmm, this is a good question.
1. You are correct, not all the subsurfaces necessary to have all of these values. I will expect at least `glazed_area` and `opaque_area`, but not `shgc`, and probably `u_factor`. - maybe .get with a default value could work well.
2. In the rule_check I think we should have a strict equality as RDS suggested. She mentioned that if comparing values such as u_factor between RMDs, then we should not allow margins.  I need to confirm this with Maria again.

Reply:
I had forgotten about your #2.  So get for all of these should work well; with no extra argument the default is None.  None == None is True so we're all set.